turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -249,14 +249,17 @@ pub trait TurboTasksBackendApi<B: Backend + 'static>: TurboTasksCallApi + Sync +
     /// The caller must ensure that the task id is not used anymore.
     unsafe fn reuse_transient_task_id(&self, id: Unused<TaskId>);
 
+    /// Schedule a task for execution.
     fn schedule(&self, task: TaskId);
-    fn schedule_backend_background_job(&self, job: B::BackendJob);
-    fn schedule_backend_foreground_job(&self, job: B::BackendJob);
 
-    fn try_foreground_done(&self) -> Result<(), EventListener>;
-    fn wait_foreground_done_excluding_own<'a>(
-        &'a self,
-    ) -> Option<Pin<Box<dyn Future<Output = ()> + Send + 'a>>>;
+    /// Schedule a primary backend job for execution.
+    fn schedule_backend_primary_job(&self, job: B::BackendJob);
+
+    /// Schedule a secondary backend job for execution.
+    ///
+    /// Secondary jobs are not counted towards activeness of the system. The system is considered
+    /// idle even with active secondary jobs.
+    fn schedule_backend_secondary_job(&self, job: B::BackendJob);
```

Comment:
foreground vs background seem like more intuitive names... 

i see that you didn't like the old definition of a background job, but the naming seems better.

Reply:
foreground vs background sounds like it imply priority between these jobs, which is not true.

Reply:
I changed it back to foreground and background